### PR TITLE
Add Kick Command

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: gomint-server/src/main/java/io/gomint/server/command/internal/KickCommand.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import io.gomint.entity.EntityPlayer;
+
+import java.util.Map;
+
+@Name("kick")
+@Description("Kick a player")
+@Permission("gomint.commands.kick")
+@Overload({
+        @Parameter(name = "target", validator = TargetValidator.class),
+        @Parameter(name = "reason", validator = StringValidator.class, arguments = {"[a-zA-Z0-9ß\\-]+"})
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/command/internal/KickCommand.java]: reason should be a text validator so you can enter texts with spaces

File: gomint-server/src/main/java/io/gomint/server/command/internal/KickCommand.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+})
+@Overload({
+        @Parameter(name = "target", validator = TargetValidator.class),
+})
+
+public class KickCommand extends Command {
+
+    @Override
+    public CommandOutput execute(EntityPlayer entityPlayer, String s, Map<String, Object> map) {
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/command/internal/KickCommand.java]: please change the signature of the method to the new API (CommandSender sender, String alias, Map<String, Object> arguments) so you can kick from console.

**Review by geNAZt — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
`<dependency>
            <groupId>io.gomint</groupId>
            <artifactId>gomint-api</artifactId>
            <version>1.0.0-SNAPSHOT</version>
        </dependency>` doesn't show CommandSender

**Comment by geNAZt** [comment]
Yeah sorry, updated maven just now. please refresh your project

**Comment by ghost** [comment]
is this a no good?

File: gomint-server/src/main/java/io/gomint/server/command/internal/KickCommand.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import io.gomint.entity.EntityPlayer;
+
+import java.util.Map;
+
+@Name("kick")
+@Description("Kick a player")
+@Permission("gomint.commands.kick")
+@Overload({
+        @Parameter(name = "target", validator = TargetValidator.class),
+        @Parameter(name = "reason", validator = TextValidator.class, arguments = {"[a-zA-Z0-9ß\\-]+"})
```
Comment (Janmm14) [inline gomint-server/src/main/java/io/gomint/server/command/internal/KickCommand.java:18]: maybe mark it optional and remove second overload

File: gomint-server/src/main/java/io/gomint/server/command/internal/KickCommand.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    public CommandOutput execute(CommandSender player, String alias, Map<String, Object> arguments) {
+        CommandOutput output = new CommandOutput();
+
+        EntityPlayer target = (EntityPlayer) arguments.get("target");
+        String reason = (String) arguments.get("reason");
+
+        if (target == null) {
+            return output.fail("Invalid argument given. Usage: /kick <player> <reason>");
+        }
+        if (arguments.containsKey("target") && !arguments.containsKey("reason")) {
```
Comment (Janmm14) [inline gomint-server/src/main/java/io/gomint/server/command/internal/KickCommand.java:36]: `arguments.containsKey("target")` is always true by now

**Comment by lukeeey** [comment]
Added in #337 (https://github.com/GoMint/GoMint/commit/b982106360623a9f43f0edbfa7c1d2b96944faaa)

**Pull request closed without merging**